pkg/provider/azure_zones.go

```diff
@@ -31,6 +32,58 @@ import (
 	"sigs.k8s.io/cloud-provider-azure/pkg/consts"
 )
 
+func (az *Cloud) refreshZones(refreshFunc func()) {
+	ticker := time.NewTicker(consts.ZoneFetchingInterval)
+	defer ticker.Stop()
+
+	az.regionZonesMap = make(map[string][]string)
+	refreshFunc()
+	for ; true; <-ticker.C {
+		refreshFunc()
+	}
+}
+
+func (az *Cloud) getRegionZonesMap() {
+	klog.V(4).Infof("refreshZones: starting to fetch all available zones for the subscription %s", az.SubscriptionID)
+	zones, rerr := az.ZoneClient.GetZones(context.Background(), az.SubscriptionID)
+	if len(zones) == 0 || rerr != nil {
+		klog.Warningf("refreshZones: error when get zones, will retry after %s", consts.ZoneFetchingInterval.String())
+		return
+	}
+
+	az.updateRegionZonesMap(zones)
+}
+
+func (az *Cloud) updateRegionZonesMap(zones map[string][]string) {
+	az.refreshZonesLock.Lock()
+	defer az.refreshZonesLock.Unlock()
+
+	for region, z := range zones {
+		az.regionZonesMap[region] = z
+	}
+}
+
+func (az *Cloud) getRegionZonesBackoff(region string) ([]string, error) {
+	if len(az.regionZonesMap) != 0 {
+		return az.regionZonesMap[region], nil
+	}
+
+	klog.V(2).Infof("getRegionZonesMapWrapper: the region-zones map is not initialized successfully, retrying immediately")
+	for i := 0; i < az.RefreshZonesRetryCount; i++ {
+		klog.V(6).Infof("getRegionZonesMapWrapper: retrying, times %d", i+1)
+		zones, rerr := az.ZoneClient.GetZones(context.Background(), az.SubscriptionID)
+		if len(zones) == 0 || rerr != nil {
+			klog.Warningf("getRegionZonesMapWrapper: failed to fetch zones information: %v", rerr.Error())
+			continue
```

Comment:
should wait a while between each retry, e.g. sleep a few seconds (2^i seconds)

Reply:
ok, I will use wait.ExponentialBackoff